### PR TITLE
skhd: 0.3.5 -> 0.3.9

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Bumping skhd version to latest. 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — APPROVED** [review]
I took the liberty to update a couple of other things.

LGTM now.

**Comment by drupol** [comment]
Looks like the build on `x86_64-darwin` is failing.

**Comment by khaneliman** [comment]
> Looks like the build on `x86_64-darwin` is failing. Could it be a missing `libiconv` in `nativeBuildInputs` ?
Spun up a x86_64-darwin vm and this is what I get. 

image

**Comment by drupol** [comment]
I think we have to call the @NixOS/darwin-maintainers here :)

**Comment by khaneliman** [comment]
This is the diff from 0.3.5 -> 0.3.9 https://github.com/koekeishiya/skhd/compare/v0.3.5...v0.3.9

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -3941,7 +3941,7 @@ with pkgs;
   reattach-to-user-namespace = callPackage ../os-specific/darwin/reattach-to-user-namespace { };
 
   skhd = callPackage ../os-specific/darwin/skhd {
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:3943]:
```suggestion
  skhd = darwin.apple_sdk_11_0.callPackage ../os-specific/darwin/skhd {
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -3941,7 +3941,7 @@ with pkgs;
...
-    inherit (darwin.apple_sdk.frameworks) Carbon;
+    inherit (darwin.apple_sdk.frameworks) Carbon Cocoa;
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:3944]:
```suggestion
    inherit (darwin.apple_sdk_11_0.frameworks) Carbon Cocoa;
```

**Comment by khaneliman** [comment]
Looks like that worked! I'll update PR real quick.

**Comment by drupol** [comment]
@GrahamcOfBorg build skhd

File: pkgs/os-specific/darwin/skhd/default.nix [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
-  meta = with lib; {
+  meta = {
     description = "Simple hotkey daemon for macOS";
     homepage = "https://github.com/koekeishiya/skhd";
-    platforms = platforms.darwin;
-    maintainers = with maintainers; [ cmacrae lnl7 periklis ];
-    license = licenses.mit;
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ cmacrae lnl7 periklis khaneliman];
+    platforms = lib.platforms.darwin;
```
Comment (drupol) [inline pkgs/os-specific/darwin/skhd/default.nix:39]: I think there's an issue here

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/darwin/skhd/default.nix (hunk 3 above)
Comment (khaneliman) [reply pkgs/os-specific/darwin/skhd/default.nix:39]: what's that?

Comment (drupol) [reply pkgs/os-specific/darwin/skhd/default.nix:39]: I think you should update `meta.platforms` to support Linux and Darwin.

Comment (emilazy) [reply pkgs/os-specific/darwin/skhd/default.nix:39]: It's macOS-only software.

Comment (khaneliman) [reply pkgs/os-specific/darwin/skhd/default.nix:39]: This is in the os-specific/darwin though, no?

Comment (drupol) [reply pkgs/os-specific/darwin/skhd/default.nix:39]: Oh... My bad... Reviewing too much PRs :)

**Comment by khaneliman** [comment]
Appreciate the help, thanks. :)

**Pull request merged**